int32 estartt
int32 estopt 
---
int32 donet